app/template_folder/rest.py

```diff
@@ -89,3 +92,57 @@ def delete_template_folder(service_id, template_folder_id):
     dao_delete_template_folder(template_folder)
 
     return '', 204
+
+
+@template_folder_blueprint.route('/move-to-folder', methods=['POST'])
+@template_folder_blueprint.route('/move-to-folder/<uuid:target_template_folder_id>', methods=['POST'])
+@transactional
+def move_to_template_folder(service_id, target_template_folder_id=None):
+    data = request.get_json()
+
+    validate(data, post_move_template_folder_schema)
+
+    if target_template_folder_id:
+        target_template_folder = dao_get_template_folder_by_id_and_service_id(target_template_folder_id, service_id)
+    else:
+        target_template_folder = None
+
+    for template_folder_id in data['folders']:
+        try:
+            template_folder = dao_get_template_folder_by_id_and_service_id(template_folder_id, service_id)
+        except NoResultFound:
+            msg = 'Could not move to folder: No folder found with id {} for service {}'.format(
+                template_folder_id,
+                service_id
+            )
+            current_app.logger.error(msg)
+            raise InvalidRequest(msg, status_code=400)
+
+        if target_template_folder and template_folder.is_parent_of(target_template_folder):
```

Comment:
I don't know if this belongs in `is_parent_of` or as a separate check, but we should also check that we're not making the folder its own parent (so `target_template_folder_id != template_folder_id`)